qiskit_experiments/library/calibration/fine_frequency_cal.py

```diff
@@ -0,0 +1,146 @@
+        backend: Optional[Backend] = None,
+        delay_duration: Optional[int] = None,
+        repetitions: List[int] = None,
+        auto_update: bool = True,
+        gate_name: str = "sx",
+    ):
+        r"""see class :class:`FineFrequency` for details.
+
+        Note that this class implicitly assumes that the target angle of the gate
+        is :math:`\pi` as seen from the default analysis options.
+
+        Args:
+            qubit: The qubit for which to run the fine frequency calibration.
+            calibrations: The calibrations instance with the schedules.
+            backend: Optional, the backend to run the experiment on.
+            delay_duration: The duration of the delay at :math:`n=1`. If this value is
+                not given then the duration of the gate named ``gate_name`` in the
+                calibrations will be used.
+            auto_update: Whether or not to automatically update the calibrations. By
+                default this variable is set to True.
+            gate_name: This argument is only needed if ``delay_duration`` is None. This
+                should be the name of a valid schedule in the calibrations.
+        """
+        if delay_duration is None:
+            delay_duration = calibrations.get_schedule(gate_name, qubit).duration
+
+        super().__init__(
+            calibrations,
+            qubit,
+            delay_duration=delay_duration,
+            schedule_name=None,
+            repetitions=repetitions,
+            backend=backend,
+            cal_parameter_name=calibrations.__drive_freq_parameter__,
+            auto_update=auto_update,
+        )
+
+        self.set_transpile_options(inst_map=calibrations.default_inst_map)
+
+        if self.backend is not None:
+            self.set_experiment_options(dt=getattr(self.backend.configuration(), "dt", None))
+
+    @classmethod
+    def _default_experiment_options(cls):
+        """default values for the fine frequency calibration experiment.
+
+        Experiment Options:
+            dt (float): The duration of the samples of the arbitrary waveform generator that
+                is used to generate the pulses.
+        """
+        options = super()._default_experiment_options()
+        options.dt = None
+        return options
+
+    def _add_cal_metadata(self, experiment_data: ExperimentData):
+        """Add metadata to the experiment data making it more self contained.
+
+        The following keys are added to the experiment's metadata:
+            cal_param_value: The value of the drive frequency parameter. This value together with
+                the fit result will be used to find the new value of the drive frequency parameter.
+            cal_param_name: The name of the parameter in the calibrations.
+            cal_group: The calibration group to which the parameter belongs.
+            delay_duration: The duration of the first delay.
+            dt: The number of AWG samples of the delay.
```

Comment:
Minor question: I don't see AWG or arbitrary waveform generator used in qiskit-experiments other than in this PR. I wonder if it is better to leave that out and just refer to "samples." Or is there an alternative descriptor for the samples? The two reasons to avoid using AWG are that it might be confusing to someone less familiar with backend hardware and that it might not necessarily be accurate (does dt always correspond to AWG samples or could there be an abstraction layer in between?).

Reply:
Done here: f10a02e